pylint/checkers/utils.py

```diff
@@ -1719,3 +1719,19 @@ def get_node_first_ancestor_of_type(
         if isinstance(ancestor, ancestor_type):
             return ancestor
     return None
+
+
+def get_node_first_ancestor_of_type_and_its_child(
+    node: nodes.NodeNG, ancestor_type: Union[Type[T_Node], Tuple[Type[T_Node]]]
+) -> Tuple[Optional[T_Node], Optional[T_Node]]:
```

Comment:
I need to spruce up my `mypy` knowledge but I think this is incorrect and should be:
`-> Tuple[Optional[T_Node], Optional[K_Node]]:`.

Using `T_Node` twice makes it so that both should be the same type. I think this would be correct:
`Tuple[Optional[T_Node], Optional[nodes.NodeNG ]]`.

It might even better to do:
`Union[Tuple[None, None]. Tuple[T_Node, nodes.NodeNG ]]` as that shows that `None` should always pair with `None`